src/scriptInterfaceMagic.h

```diff
@@ -196,7 +197,11 @@ struct convert<P<T>>
             P<PObject>** p = static_cast< P<PObject>** >(lua_newuserdata(L, sizeof(P<PObject>*)));
             *p = new P<PObject>();
             (**p) = ptr;
-            lua_settable(L, -3);
+
+            // protect the new userdata's metatable
+            protectLuaMetatable(L);

```

Comment:
I think this protection can be done here: https://github.com/daid/SeriousProton/blob/master/src/scriptInterfaceMagic.h#L606
As soon as the metatable for each class is made. And then we don't need to do any protection here.